### ISSUE TITLE
Cannot get qMsgs if the WPA deleted and re-created

### ISSUE THREAD

**Issue by KangBK0120** [practo/k8s-worker-pod-autoscaler#134]
Hi, I faced an issue with the controller in the ```master``` branch.

If I create a dummy deployment and a dummy WPA, it worked without any problem. 
The deployment does not do any job from SQS. It does not receive or process the messages in the queue. And the dummy WPA simply auto-scales it.

However, the controller failed to get qMsgs if I deleted the WPA and re-create it with the same YAML.

![image](https://user-images.githubusercontent.com/25279765/121871312-ed6de200-cd3e-11eb-8cb1-7bd686fe6c37.png)

Here are the YAML files I used

```.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: dummy-deployment
spec:
  selector:
    matchLabels:
      app: dummy-deployment
  replicas: 1
  template:
    metadata:
      labels:
        app: dummy-deployment
    spec:
      containers:
      - name: nginx
        image: nginx:1.7.9
        ports:
        - containerPort: 80
```

```.yaml
apiVersion: k8s.practo.dev/v1
kind: WorkerPodAutoScaler
metadata:
  name: dummy-wpa
spec:
  minReplicas: 1
  maxReplicas: 100
  deploymentName: dummy-deployment
  queueURI: https://sqs.{region}.amazonaws.com/{id}/dummy-queue
  targetMessagesPerWorker: 1
  maxDisruption: "100%"
```
 I did not change anything in the workerpodautoscaler deployment.

**Comment by KangBK0120** [comment]
I found that the issue comes from poller.go

When a new WPA resource is created, the thread is successfully created and works fine.
However, if a user deletes the WPA, the thread is deleted but the sync function in poller still checks and holds its status.

And if a new WPA is created with the same key, the thread is not successfully created and thus qMsgs cannot be fetched.

Therefore the Sync logic should be like follows.

```go
func (p *Poller) Sync(stopCh <-chan struct{}) {
	for {
		select {
		case listResultCh := <-p.listThreadCh:
			listResultCh <- DeepCopyThread(p.threads)
		case threadStatus := <-p.updateThreadCh:
			for key, status := range threadStatus {
				if status == false {
					delete(p.threads, key)
				} else {
					p.threads[key] = status
				}
			}
		case <-stopCh:
			klog.V(1).Info("Stopping sync thread of poller gracefully.")
			return
		}
	}
}
```

**Comment by KangBK0120** [comment]
### Before: 
![image](https://user-images.githubusercontent.com/25279765/121987664-e3e28980-cdd3-11eb-9951-e427c5fa70f9.png)
![image](https://user-images.githubusercontent.com/25279765/121987691-f1980f00-cdd3-11eb-89e7-b8d74147227b.png)

### After: 
![image](https://user-images.githubusercontent.com/25279765/121987729-04aadf00-cdd4-11eb-8294-0750067ac7cf.png)

Sorry for the bad output format. I'm new to Go

**Comment by alok87** [comment]
Thanks @KangBK0120 for reporting, debugging and fixing the issue. 
I have released a patch release [v1.4.1](https://github.com/practo/k8s-worker-pod-autoscaler/releases/tag/v1.4.1) for it as it could be critical for others if they recreate with the same WPA name. 

```
practodev/workerpodautoscaler:v1.4.1
```
